### DISCUSSION TITLE
ST75256 artifacts

### DISCUSSION THREAD

**Discussion by dynek** [olikraus/u8g2#1846] [Q&A] [answered]
Hello everyone,

I'm using the full buffer mode, as explained in the tutorial, just to print a few things on a the LCD, I'll improve this later.
Problem is, there are some artifacts quite often when screen is being refreshed.

**When OK**
![photo1](https://user-images.githubusercontent.com/591091/164396799-65ed48a2-e6e4-458b-8369-f0ba6dd946fc.jpg)

**When not OK**
![photo2](https://user-images.githubusercontent.com/591091/164396824-a0787d01-1cd8-40fb-aa77-c8b6e488bbc0.jpg)

Any idea why this is happening.
I'm doing a clearbuffer, setfont and multiple drawstr in a freertos task, making sure semaphore is available to not have issues with other i2c interactions e.g. RTC and ADS.

Thank you

**Comment by olikraus** [comment]
Can you post your code, especially the code for the u8g2 screen drawing.

**Reply by dynek** [reply]
Here you go:
```
      DEBUG_PRINTLN("[updateLCD] refresh data on LCD - start");
      u8g2.clearBuffer();
      u8g2.setFont(u8g2_font_ncenB12_tr);
      //u8g2.setCursor(0, 15);
      char message[25] = {};
      sprintf(message, "%d/%d/%d %d:%d:%d", now.year(), now.month(), now.day(), now.hour(), now.minute(), now.second());
      u8g2.drawStr(0, 15, message);

      char voltageVehicle[20] = {};
      sprintf(voltageVehicle, "Vehicle: %.1f", currentVoltageVehicule);
      //u8g2.setCursor(0, 30);
      u8g2.drawStr(0, 30, voltageVehicle);

      char voltageLeisure[20] = {};
      sprintf(voltageLeisure, "Leisure: %.1f", currentVoltageLeisure);
      u8g2.drawStr(0, 45, voltageLeisure);

      char internalTemp[20] = {};
      sprintf(internalTemp, "Temp inside: %.1f", temperature_inside);
      u8g2.drawStr(0, 60, internalTemp);

      char externalTemp[20] = {};
      sprintf(externalTemp, "Temp outside: %.1f", temperature_outside);
      u8g2.drawStr(0, 75, externalTemp);

      if(!water_used_full) {
        u8g2.drawStr(0, 90, "Used water: not full");
      } else {
        u8g2.drawStr(0, 90, "Used water: FULL");
      }

      if(water_clear_top) {
        u8g2.drawStr(0, 105, "Clear water: full");
      } else if(water_clear_middle) {
        u8g2.drawStr(0, 105, "Clear water: middle");
      } else if(water_clear_bottom) {
        u8g2.drawStr(0, 105, "Clear water: BOTTOM");
      } else {
        u8g2.drawStr(0, 105, "Clear water: UNKNOWN");
      }
 
      u8g2.sendBuffer();
      DEBUG_PRINTLN("[updateLCD] refresh data on LCD - done");
```

**Comment by olikraus** [comment, marked as answer]
Code looks ok to me.
Are you sure, that the communication with the display is not interrupted somehow?
Can you try to disable any task switch and interrupts as long as sendBuffer() is executed?

**Reply by dynek** [reply]
It looks like the problem came from the fact that I was using `Wire.begin` to configure SDA, SCL pins, rather than `Wire.setPins`.
What helped is that I upgraded Espressif 32 to 4.1.0 and this started to show up, and i2c stuff wasn't working anymore:
```
[  2208][E][Wire.cpp:283] setClock(): Bus is in Slave Mode
[  2219][E][Wire.cpp:308] beginTransmission(): Bus is in Slave Mode
[  2220][E][Wire.cpp:331] endTransmission(): Bus is in Slave Mode
```
With `setPins` & semaphore everything seems to work properly, yeah!
